### PR TITLE
mle: init at 1.4.1

### PR DESCRIPTION
**Pull request by adsr**
###### Motivation for this change

Small, flexible, terminal-based text editor

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by teto** [comment]
@GrahamcOfBorg build mle

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -184,6 +184,12 @@
     githubId = 1250775;
     name = "Adolfo E. García Castro";
   };
+  adsr = {
```
Comment (alexarice) [inline maintainers/maintainer-list.nix]: The addition to maintainers.nix should be in a separate commit titled "maintainers: add adsr"

File: pkgs/applications/editors/mle/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  postPatch = ''
+    substituteInPlace Makefile --replace "-llua5.3" "-llua";
+    substituteInPlace mle.h    --replace "<lua5.3/" "<";
+  '';
+
+  buildInputs = [ termbox pcre uthash lua5_3 ];
+
+  doCheck = true;
+  checkPhase = "make test";
```
Comment (alexarice) [inline pkgs/applications/editors/mle/default.nix]: I believe this should be the default checkPhase

File: pkgs/applications/editors/mle/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  src = fetchFromGitHub {
+    owner = "adsr";
+    repo = "mle";
+    rev = "v${version}";
+    sha256 = "1yz6yrnib3k7kg30v2sllc13y328faf8k3nyh1y2s3rinf5fyzs2";
+  };
+
+  postPatch = ''
+    substituteInPlace Makefile --replace "-llua5.3" "-llua";
```
Comment (alexarice) [inline pkgs/applications/editors/mle/default.nix]: Might be worth putting some motivation for these somewhere, did you create this fix yourself or is it from somewhere else

Comment (alexarice) [reply pkgs/applications/editors/mle/default.nix]: Have just realised this is your own project, is there any reason this can't be upstreamed?

Comment (adsr) [reply pkgs/applications/editors/mle/default.nix]: The version in trunk (`-llua5.3` / `<lua5.3/lua.h>`) works with Debian-based systems. Alternatives to this are using `pkg-config` or trying to guess the location of Lua dynamically. Personally I prefer this small patch as I find it the simplest to read.
